srcts/src/components/darkModeSwitch.ts

```diff
@@ -0,0 +1,298 @@
+import type { CSSResultGroup, LitElement } from "lit";
+import { css, html } from "lit";
+import { property } from "lit/decorators.js";
+import { BslibElement } from "./webcomponents/_bslibElement";
+
+// Inspired by:
+// https://web.dev/building-a-theme-switch-component/
+// https://web.dev/patterns/theming/theme-switch/
+// https://github.com/argyleink/gui-challenges/tree/main/theme-switch
+
+type ToggleMessage = {
+  method: "toggle";
+  value: DarkModeSwitch["themeValue"] | "toggle";
+};
+
+export class DarkModeSwitch extends BslibElement {
```

Comment:
```suggestion
// This class extends BslibElement only as a convenience for registering the custom element 
// and Shiny input binding. All the logic here (besides the `isShinyInput`/`tagName` fields) is 
// designed to work without Bootstrap/bslib
export class DarkModeSwitch extends BslibElement {
```

Reply:
FYI I started this PR with the original implementation in the first commit, so we can use this PR's history, see 6653f5e600d697e6a7cb2ccdde472e3e863e43ec